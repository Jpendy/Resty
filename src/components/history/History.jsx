import React from 'react';
import PropTypes from 'prop-types';

const History = ({ historyArray }) => {
    
  const historyList = historyArray.map(item => {
    return (
      <li key={Math.random()}>
        {`Endpoint: ${item.endPoint} - Method: ${item.method} `}
      </li>
    );
  });

  return (
    <>
      <h3>History:</h3>
      <ul>
        {historyList}
      </ul>
    </>
  );
};

History.propTypes = {
  historyArray: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default History;
