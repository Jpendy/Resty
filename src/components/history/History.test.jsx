import React from 'react';
import { shallow } from 'enzyme';
import History from './History';

describe('History component', () => {
  it('renders History', () => {
    const wrapper = shallow(<History historyArray={[{ endPoint: 'end point 1', method: 'GET' }, { endPoint: 'end point 2', method: 'PUT' }]}  />);
    expect(wrapper).toMatchSnapshot();
  });
});
