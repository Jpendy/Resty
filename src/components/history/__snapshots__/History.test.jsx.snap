// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<Fragment>
  <h3>
    History:
  </h3>
  <ul>
    <li
      key="0.9879073260333533"
    >
      Endpoint: end point 1 - Method: GET 
    </li>
    <li
      key="0.9855736165547888"
    >
      Endpoint: end point 2 - Method: PUT 
    </li>
  </ul>
</Fragment>
`;
